tests/test_endpoint.py

```diff
@@ -0,0 +1,140 @@
+"""
+LSST Data Management System
+See COPYRIGHT file at the top of the source tree.
+
+This product includes software developed by the
+LSST Project (http://www.lsst.org/).
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the LSST License Statement and
+the GNU General Public License along with this program. If not,
+see <http://www.lsstcorp.org/LegalNotices/>.
+"""
+from __future__ import absolute_import, division, print_function
+import unittest
+
+import numpy as np
+import astshim
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+
+def makeRawArrData(nPoints, nAxes, delta=0.123):
+    return np.array([[i + j*delta for j in range(nAxes)] for i in range(nPoints)])
+
+def makeRawPointData(nAxes, delta=0.123):
+    return [i*delta for i in range(nAxes)]
+
+class EndpointTestCase(unittest.TestCase):
+
+    def testSpherePointEndpoint(self):
+        endpoint = afwGeom.SpherePointEndpoint()
+        self.checkEndpointBasics(endpoint=endpoint, pointType=afwGeom.SpherePoint, nAxes=2)
+
+        for doPermute in (False, True):
+            frame = astshim.SkyFrame()
+            if doPermute:
```

Comment:
I'm going to push back here. I think the code is more readable as written. I worry that by moving it to a subroutine it will be harder to see what's going on and far more verbose for so few lines of code. I could probably use variables to and always call permute (a no-op in one case and permutation in the other), but I fear the results will be harder to read.